solutionbox/structured_data/datalab_solutions/structured_data/trainer/util.py

```diff
@@ -91,6 +105,26 @@ def end(self, session=None):
       _copy_all(self._additional_assets, assets_dir)
 
 
+# ==============================================================================
+# Reading the input csv files and parsing its output into tensors.
+# ==============================================================================
+
+def parse_example_tensor(examples, train_config):
+
+  # record_defaults are used by tf.decode_csv to insert defaults, and to infer
+  # the datatype. 
+  record_defaults = [[train_config['csv_defaults'][name]] 
```

Comment:
I have to have a default value when reading data from csv in TF, so there is no way of knowing at training time if data is missing. Picking a bad default value would make a model useless, so I would want to just ask the user for these values.